src/lib/storage/reference_segment/reference_segment_iterable.hpp

```diff
@@ -71,12 +70,9 @@ class ReferenceSegmentIterable : public SegmentIterable<ReferenceSegmentIterable
           if constexpr (std::is_same_v<SegmentType, LZ4Segment<T>>) return;
 
           if constexpr (!std::is_same_v<SegmentType, ReferenceSegment>) {
-            auto accessor = std::make_shared<SegmentAccessor<T, SegmentType>>(typed_segment);
-
-            auto begin = SingleChunkIterator<SegmentAccessor<T, SegmentType>>{accessor, begin_it, begin_it};
-            auto end = SingleChunkIterator<SegmentAccessor<T, SegmentType>>{accessor, begin_it, end_it};
+            const auto segment_iterable = create_iterable_from_segment<T>(typed_segment);
+            segment_iterable.with_iterators(pos_list, functor);
```

Comment:
This is the actual change. The rest is just moved code to deal with header include orders :/

Reply:
This is a great idea.
But I have to ask you on Wednesday why this only works for `EraseReferencedSegmentType::No`.

Reply:
To be honest, I didn't look that far. With erased segment types, we basically declare that we stopped caring about performance anyway. I don't think that there is a general reason why this shouldn't work.

Let's see what CI says about it.